src/server/saved_objects/saved_objects_mixin.js

```diff
@@ -30,30 +30,34 @@ export function savedObjectsMixin(kbnServer, server) {
     const adminCluster = server.plugins.elasticsearch.getCluster('admin');
 
     try {
-      await adminCluster.callWithInternalUser('cluster.health', {
-        timeout: `${server.config().get('savedObjects.indexCheckTimeout')}ms`,
-        index: server.config().get('kibana.index'),
-        waitForStatus: 'yellow',
+      const index = server.config().get('kibana.index');
+      await adminCluster.callWithInternalUser('indices.putTemplate', {
```

Comment:
__question:__ it must be a dumb question, but why do we want to put this template on every write? As far as I understand `server.getKibanaIndexMappingsDsl()` is supposed to return the same thing most of the time (always?) or am I missing something?

Reply:
Not a dumb question. My intention was to ensure that Elasticsearch has the index template before every write so that even if a user deletes the index and index template in elasticsearch, writes will still work (unless the index template is deleted in the tiny amount of time between putting the index template and indexing the document).

Reply:
Thanks for the explanation! Then __optional nit:__ maybe you can add this comment to the code or even move this piece of code to a dedicated function like `ensureKibanaIndexExists` or something like that to make intention clearer (your comment above still seems to be useful to have there as well)?